client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -477,6 +479,65 @@ export class PluginsList extends Component {
 		}
 	};
 
+	updateAllPluginsDialog = () => {
+		const { pluginUpdateCount, translate } = this.props;
+
+		const siteDetails = this.props.selectedSiteSlug ? this.props.selectedSiteSlug : 'all sites';
+
+		const dialogOptions = {
+			additionalClassNames: 'plugins__confirmation-modal',
+		};
+
+		const heading =
+			pluginUpdateCount === 1
+				? translate( 'Update 1 plugin' )
+				: translate( 'Update %(pluginUpdateCount)d plugins', { args: { pluginUpdateCount } } );
+
+		const combination =
+			( ! siteDetails ? 'n sites' : '1 site' ) +
+			' ' +
+			( pluginUpdateCount > 1 ? 'n plugins' : '1 plugin' );
+
+		let message = 'You are about to update all plugins on all sites';
+		switch ( combination ) {
+			case '1 site 1 plugin':
+				message = translate( ' You are about to update 1 plugin on %(siteDetails)s', {
+					args: { siteDetails },
+				} );
+				break;
+			case '1 site n plugins':
+				message = translate(
+					' You are about to update %(pluginUpdateCount)d plugins on %(siteDetails)s',
+					{
+						args: {
+							pluginUpdateCount: pluginUpdateCount,
+							siteDetails: siteDetails,
+						},
+					}
+				);
+				break;
+			case 'n sites 1 plugin':
+				message = translate( ' You are about to update 1 plugin on all sites', {
+					args: { siteDetails },
+				} );
+				break;
+			case 'n sites n plugins':
+				message = translate(
+					' You are about to update %(pluginUpdateCount)d plugins on all sites',
+					{ args: { pluginUpdateCount } }
+				);
+				break;
+		}
+
+		acceptDialog(
```

Comment:
I notice that this is still not following the new design. It currently looks like the below:

<img width="413" alt="Screenshot 2022-09-20 at 4 14 17 PM" src="https://user-images.githubusercontent.com/10586875/191238099-e1822c9b-c960-49b0-bc05-c4717b74641a.png">

You can see how it is done for other actions(Activate, Deactivate, etc.)

We could reuse the method - `getPluginActionDailogMessage` to get the message and we can also get rid of the switch case here.